webapp/src/test/java/info/rmapproject/webapp/service/DataDisplayServiceImplTest.java

```diff
@@ -201,6 +202,26 @@ public void testGetResourceBatch() throws Exception {
 	}
 	
 	/**
+	 * Basic check that getResourceBatch returns not found if id is not available.
+	 * @throws Exception
+	 */
+	@Test
+	public void testGetResourceBatchWhenNoMatches() throws Exception {
+		// agent already exists, so create a disco
+		ORMapDiSCO disco = getRMapDiSCOObj(TestFile.DISCOB_V1_XML);
+		String discoUri = disco.getId().toString();
+        assertNotNull(discoUri);
+		rmapService.createDiSCO(disco, requestAgent);
+		String uriInDisco = "fakefake:uri";
+		try {
+			dataDisplayService.getResourceBatch(uriInDisco, 0, PaginatorType.RESOURCE_GRAPH); //graph excludes literals
+			fail("Should have thrown an RMapObjectNotFoundException");
+		} catch (RMapObjectNotFoundException ex) {
```

Comment:
Don't need to change this, but in the case where there is no testing logic after catching an expected exception, you could optionally annotate the method with the `expected` attribute of `@Test`:
`@Test(expected = RMapObjectNotFoundException.class)` and eliminate the try/catch block and `fail` assertion.

Reply:
another good tip! Thanks!!